### ISSUE TITLE
Different length sort behavior against isort

### ISSUE THREAD

**Issue by yanyongyu** [astral-sh/ruff#9870]
I'm trying to migrate isort and black to ruff but encountered some problems when using ruff's isort configs.

Ruff version 0.2.1.

Suppose I have a first-party package named `package` and the test code is:

```python
from package.utils import a
from package.func import function
```

with the ruff config:

```toml
[tool.ruff.lint]
select = ["I"]

[tool.ruff.lint.isort]
length-sort = true
force-sort-within-sections = true
```

ruff will report import order error and fix it into:

```python
from package.func import function
from package.utils import a
```

but isort sorts it as the origin test code given above.

I make one more test, renaming the `package.func` to `package.function`:

```python
from package.utils import a
from package.function import function
```

this time ruff does not report any error. It seems ruff sort the import order by module name length instead of import expression length?

**Comment by charliermarsh** [comment]
Can you clarify for me -- if I run `isort` over:

```python
from package.utils import a
from package.func import function
```

With `length_sort = true`, it rewrites to:

```python
from package.func import function
from package.utils import a
```

In my testing, Ruff has the same behavior. But it seems like you're seeing different behavior with isort?

**Comment by yanyongyu** [comment]
i sort the code with:

```bash
isort --profile black --length-sort --force-sort-within-sections package/__init__.py
```

**Comment by yanyongyu** [comment]
Or use the pyproject.toml to config isort:

```toml
[tool.ruff.lint]
select = ["I"]

[tool.ruff.lint.isort]
length-sort = true
force-sort-within-sections = true

[tool.isort]
profile = "black"
length_sort = true
force_sort_within_sections = true
```

then sort the imports with `isort .`

the output code is:

```python
from package.utils import a
from package.func import function
```

**Comment by charliermarsh** [comment]
It this an isort bug? Removing `--force-sort-within-sections` shows the same behavior as Ruff, but I don't see why `--force-sort-within-sections ` should have an effect on the sort in this case (both imports are in the same section).

**Comment by yanyongyu** [comment]
> Isort docs: Instead, sort the imports by module, independent of import style.

I'm not sure which behavior is correct 😂

**Comment by charliermarsh** [comment]
Haha :)

I looked through the isort source. It looks like if `--force-sort-within-sections` is set, it sorts all lines by their exact length. If it's not set, it sorts by the length of the module, then sorts each member by length. So the behavior is totally different when that flag is set. I find it confusing... But if we don't support it, we may want to document it at least.

**Comment by tomprimozic** [comment]
I'd also want a "sort imports by line length exactly" with no extra exceptions. `isort` does (almost) support that (except some edge cases / bugs) using `length_sort` and `force_sort_within_sections` options, it would be great if `ruff` did as well.

For the time being, I have to use 2 tools :/

**Comment by yanyongyu** [comment]
@charliermarsh Is there any decision about this issue? Currently, i also have to use 2 tools (isort + ruff) to lint and format the code. i hope this feature can be part of this excellent project. :pray:

**Comment by rainzee** [comment]
This is really the last hurdle for me to fully use ruff, now I have to use both tools, hopefully some progress will be made, please!🙏

**Comment by forchannot** [comment]
any progress on this issue?

**Comment by KimigaiiWuyi** [comment]
any progress?

Due to differences in sorting behavior between `ruff` and `isort`, the project code becomes messy when using them together. I would like to make this a feature request to keep `ruff`'s sorting consistent with `isort`

#14136

**Comment by MichaReiser** [comment]
We don't recommend using isort with ruff. Use one or the other
